code/search/src/fibonacci_search/fibonacci_search.py

```diff
@@ -2,35 +2,15 @@
 Part of Cosmos by OpenGenus Foundation
 '''
 
+memo = {}
 
-def fibonacci_search(arr, x):
-    length = len(arr)
-    fib_m2 = 0
-    fib_m1 = 1
-    fib_m = fib_m2 + fib_m1
 
-    while fib_m < length:
-        fib_m2 = fib_m1
-        fib_m1 = fib_m
-        fib_m = fib_m2 + fib_m1
-
-    offset = -1
-    while fib_m > 1:
-        i = min(offset + fib_m2, length - 1)
-
-        if arr[i] < x:
-            fib_m = fib_m1
-            fib_m1 = fib_m2
-            fib_m2 = fib_m - fib_m1
-            offset = i
-        elif arr[i] > x:
-            fib_m = fib_m2
-            fib_m1 = fib_m1 - fib_m2
-            fib_m2 = fib_m - fib_m1
-        else:
-            return i
-
-    if fib_m1 and offset < length - 1 and arr[offset + 1] == x:
-        return offset + 1
-
-    return -1
+def fib(n):
```

Comment:
Instead of updating the current file, please consider moving this function to another file (maybe `fibonacci_search_memoized.py`.